src/unix-manager.c

```diff
@@ -700,18 +702,112 @@ TmEcode UnixManagerCaptureModeCommand(json_t *cmd,
     SCReturnInt(TM_ECODE_OK);
 }
 
-TmEcode UnixManagerReloadRules(json_t *cmd, json_t *server_msg, void *data)
+TmEcode UnixManagerReloadRulesWrapper(json_t *cmd, json_t *server_msg, void *data, int wait)
 {
     SCEnter();
-    DetectEngineReloadStart();
+    int r = DetectEngineReloadStart();
 
-    while (DetectEngineReloadIsDone() == 0)
-        usleep(100);
+    if (wait) {
+        while (DetectEngineReloadIsDone() == 0)
+            usleep(100);
+    } else {
+        if (DetectEngineReloadIsDone() == 0 && r == -1) {
```

Comment:
If the function is called by 'reload-rules-non-blocking' command, we won't be able to do a rule reload again because DetectEngineReloadIsDone() is not called (detect_sync.state never set to IDLE).
We should be in error case only if DetectEngineReloadIsDone() returns 0 and DetectEngineReloadStart returns -1 (live reload is already occurring)